include/nil/blueprint/parser.hpp

```diff
@@ -666,12 +664,28 @@ namespace nil {
                     frame.vectors[c] = result_vector;
                 } else if (auto expr = llvm::dyn_cast<llvm::ConstantExpr>(c)) {
                     for (int i = 0; i < expr->getNumOperands(); ++i) {
-                        put_constant(expr->getOperand(i), frame);
+                        llvm::Value *op = expr->getOperand(i);
+                        if (llvm::isa<llvm::GlobalValue>(op)) {
+                            frame.scalars[op] = globals[op];
+                        } else {
+                            put_constant(expr->getOperand(i), frame);
+                        }
                     }
                     switch (expr->getOpcode()) {
                     case llvm::Instruction::PtrToInt:
                         handle_ptrtoint(expr, expr->getOperand(0), frame);
                         break;
+                    case llvm::Instruction::GetElementPtr: {
+                        std::vector<int> gep_indices;
+                        for (unsigned i = 2; i < expr->getNumOperands(); ++i) {
+                            int gep_index = resolve_number<int>(frame, expr->getOperand(i));
+                            gep_indices.push_back(gep_index);
+                        }
+                        auto gep_res = handle_gep(expr->getOperand(0), expr->getOperand(1), expr->getOperand(0)->getType(), gep_indices, frame);
```

Comment:
`expr->getOperand(0)->getType()` gives wrong type that is not suitable to `handle_gep`

Reply:
good catch, thanks!

Reply:
done